qcodes_contrib_drivers/drivers/Bilt/ITest.py

```diff
@@ -0,0 +1,475 @@
+    
+        self.add_parameter('bilt_name',
+                           set_cmd=partial(self._parent._set_chan_name, chan_num),
+                           set_parser=str)
+
+        self.add_parameter('synchronous',
+                           get_cmd=partial(self._parent._get_chan_synchronous, chan_num),
+                           get_parser=bool,
+                           set_cmd=partial(self._parent._set_chan_synchronous, chan_num),
+                           vals=vals.Bool())
+        
+        
+        self.add_function('start',
+                          call_cmd=lambda: self._parent._set_chan_state(self.chan_num, 1))
+        
+        self.add_function('stop',
+                          call_cmd=lambda: self._parent._set_chan_state(self.chan_num, 0))
+    
+        self.bilt_name(f'Chan{chan_num:02d}')
+
+
+class iTestMultiChannelParameter(MultiChannelInstrumentParameter):
+    """
+    """
+    
+    def __init__(self, channels, param_name, *args, **kwargs):
+        super().__init__(channels, param_name, *args, **kwargs)
+
+    def trigger(self,*args,**kwargs):
+        print(args,kwargs)
+
+
+class ITest(VisaInstrument):
+    """
+    This is the QCoDeS python driver for the iTest device from Bilt.
+    """
+    
+    def __init__(self,name:str,
+                      address:str,
+                      num_chans:int=16,
+                      init_start:bool=False,
+                      synchronous:bool=True,
+                      synchronous_delay:float=1,
+                      synchronous_threshold:float=1e-5,
+                      **kwargs: Any) -> None:
+        """
+        Instantiate the instrument.
+        
+        Args:
+            name: The instrument name used by qcodes
+            address: The VISA name of the resource
+            num_chans: Number of channels to assign. Default: 16
+            init_start: If true set all channels to 0V, 1.2V range and switch
+                then on.
+            synchronous: If true, block the communication until the set voltage
+                is reached. The communication is block through a simple while loop
+                with a waiting time "synchronous_delay" at each iteration until the
+                set voltage and the measured voltage difference is below
+                "synchronous_threshold".
+            synchronous_delay: Time between to voltage measurement in second.
+            synchronous_threshold: Threshold to unblock communication in volt.
+            
+        Returns:
+            ITest object
+        """
+        super().__init__(name, address=address,
+                               terminator='\n',
+                               device_clear=False,
+                               **kwargs)
+        
+        self.idn = self.get_idn()
+        self.num_chans = num_chans
+        self.chan_range = range(1,self.num_chans+1)
+        # Fake parameter allowing to control the synchronicity of every channels
```

Comment:
is there a reason for these being init-only private attributes as opposed to proper parameters that users can change when using the instrument? just in case, a reminder, parameters can have set_cmd=None and get_cmd=None which makes them "in-memory" parameters, so they don't talk to the instrument, but still maintain a value. and then instead of maintaining the `self._synchronous` dictionary on the instrument level, this information can be conveniently maintained on the channel level

Reply:
Hum, could point.
I could change it and make in an 'in-memory' thing at the channel level but then how would I access it at the instrument level, for example here:
```python
    def _set_voltage(self, chan:int,
                           v_set:float) -> None:
        """
        Set cmd for the chXX_v parameter
        
        Args:
            chan: The 1-indexed channel number
            v_set: The target voltage
        """
        chan_id = self.chan_to_id(chan)
        self.write('{}VOLT {:.8f}'.format(chan_id, v_set))
        self.write(chan_id + 'TRIG:INPUT:INIT')
        if self._get_chan_synchronous(chan):
            v = self._get_voltage(chan)
            while abs(v_set - v)>=self._synchronous_threshold:
                sleep(self._synchronous_delay)
                v = self._get_voltage(chan)
```

Thanks.

Reply:
since the instrument is aware of all the channels and they can be accessed by their name(?), then `self.<chanX>.synchronous_delay()` will just work.

now that i look at this method, it seems to me that it actually also should go inside the channel because there is nothing in this method that depends on the instrument itself. the write/ask method calls on the channel level forward to those on the instrument level, so that you don't have to worry about